src/inner_op/somerc.rs

```diff
@@ -0,0 +1,298 @@
+#![allow(non_snake_case)]
```

Comment:
I know why this `allow(...)` is necessary, but I think a comment explaining it would be A Good Thing (TM).

We probably should have a standard wording for this. `git grep allow_non_snake_case` reveals that I have used two different wordings in the existing code:

1. allow math-like notation
2. make it possible to mimic math notation from the original paper

of which the latter is obviously the most explanatory. But since you (I suppose) have English as your mother tongue, you will probably be able to find a better way of expressing this, than this first-foreign-language speaker of (as we say in Danish) the "Denglish dialect" (where we always remember *to never split an infinitive*).

Reply:
How does `So we can use the mathematical notation from the original paper` sound to you? I can update the others if you're happy.

Reply:
Sounds good to me. Updating should probably be in another PR, so this one concentrates on `somerc` (or by rebasing and squashing all commits related to somerc into a single one, and adding a separate "update wording" commit)